dbms/src/Flash/executeQuery.cpp

```diff
@@ -200,6 +202,7 @@ std::optional<QueryExecutorPtr> executeAsPipeline(Context & context, bool intern
     if (likely(!internal))
         LOG_INFO(logger, fmt::format("Query pipeline:\n{}", executor->toString()));
     dag_context.switchToPipelineMode();
+    LocalAdmissionController::global_instance->warmupResourceGroupInfoCache(dag_context.getResourceGroupName());
```

Comment:
how about moving to the constructor of `PipelineExecutor` and moving `consumeResource` to `PipelineExecutor::execute`.

Reply:
Because resource control should only be enabled when enable_pipeline.
In the previous logic, even if the pipeline is not enabled, ru will be consumed by read_bytes